src/renderer/api-decorator.js

```diff
@@ -305,11 +309,13 @@ limitations under the License.
         }
     }
 
-    function wireUpMenu(global, options) {
+    function wireUpMenu(global) {
         global.addEventListener('contextmenu', e => {
             if (!e.defaultPrevented) {
                 e.preventDefault();
 
+                const options = getWindowOptionsSync();
+
                 if (options.contextMenu) {
                     const identity = getWindowIdentitySync();
                     syncApiCall('show-menu', {
```

Comment:
Why don't we just make the `syncApiCall` every time and handle it browser side? That way we don't have to change all this code.

Reply:
I like this, its not like we are spamming the getWindow options call. This might have been a pre-optimization.

Reply:
I agree, it would be much simpler and cleaner. And opening a window is a relatively rare event so it's not like the overhead of IPC call is a concern (barely measurable anyway). (And the sister PR could be closed.)